internal/app/cache/cache.go

```diff
@@ -36,7 +36,7 @@ type resource struct {
 }
 
 // Callback function for each eviction. Receives the key and cache value when called.
-type onEvictFunc func(key lru.Key, value interface{})
+type onEvictFunc func(key string, value resource)
```

Comment:
I think you might need to export Resource, otherwise the caller with have no access to it.